### PR TITLE
[lexical-markdown] Bug Fix: Fix nested mixed-type list structure during markdown import

### LINKED ISSUE #7938: Bug: Importing from Markdown breaks nested list structure in the editor
Editor destroys its own list structure when re-importing previously exported Markdown

This is the correct markdonw structure generated by the editor (when it’s created directly in the editor).

```
- asdasdasdasda
    1. adasdasdasdasd
        - asdasdasdasd
        - asdasdasd
        - asdasdasd
    2. asdasdasd
    3. asddddddddddddda
- asdasdasda
- asdasdasd
```

Then, if I feed this same structure back into the editor through the Markdown importer, I get the following result:

```
- asdasdasdasda

    1. adasdasdasdasd

        - asdasdasdasd
        - asdasdasd
        - asdasdasd

    1. asdasdasd
    2. asddddddddddddda

- asdasdasda
- asdasdasd
```

The structure is corrupted, even though it was originally exported correctly.
Consequently, the editor also renders everything incorrectly due to this broken import.

### PR DESCRIPTION
**Pull request by achaljhawar**
## Description

Importing markdown with nested lists of different types (e.g. bullet containing ordered) flattened them into separate top-level lists. `listReplace` only merged items into an adjacent list when types matched, so indented items of a different type couldn't merge. Fixed by allowing the merge when indented regardless of type, then correcting the nested list type after `setIndent()`.

Closes #7938 

### Before

https://github.com/user-attachments/assets/6e04e75e-44f0-48ae-995b-17f3a9f4de54

### After

https://github.com/user-attachments/assets/b2438e0a-27b8-42e3-b51c-00355cc9a0c9

### CODE REVIEW

File: packages/lexical-markdown/src/MarkdownTransformers.ts [hunk 1]
```diff
@@ -318,9 +322,12 @@ const listReplace = (listType: ListType): ElementTransformer['replace'] => {
     if (!isImport) {
       listItem.select(0, 0);
     }
-    const indent = getIndent(match[1]);
     if (indent) {
       listItem.setIndent(indent);
+      const parentList = listItem.getParent();
+      if ($isListNode(parentList) && parentList.getListType() !== listType) {
+        parentList.setListType(listType);
+      }
```
Comment (etrepum) [inline packages/lexical-markdown/src/MarkdownTransformers.ts]: This seems suspicious, what case is this for?

Comment (etrepum) [reply packages/lexical-markdown/src/MarkdownTransformers.ts]: It doesn't seem to work when the list types are mixed:

```markdown
1. item one
2. item two
   - sublist
   1. sublist
   - [ ] checklist
```

1. item one
2. item two
   - sublist
   1. sublist
   - [ ] checklist

**Comment by potatowagon** [comment]
## Review: Fix Nested Mixed-Type List Structure During Markdown Import

**Reviewed by: Navi (AI review assistant for @potatowagon)**

### Summary
Fixes markdown import to correctly handle nested lists of different types (e.g., unordered list with ordered sub-items, or ordered list with bullet sub-items). Previously, indented items of a different list type would be incorrectly merged into the parent list rather than creating a properly nested sub-list.

### What I Verified
- **Logic correctness**: The key change moves `const indent = getIndent(match[1])` earlier and adds a new branch: when `indent > 0` and the previous sibling is a list, the item is appended to that list (preserving nesting). Then, when setting indent, it checks if the parent list type differs from the item's type and restructures accordingly — creating a new nested list of the correct type.
- **Edge cases covered**: Tests include bullet→ordered, ordered→bullet, 3-level deep nesting (bullet→ordered→bullet), and mixed consecutive sublists. All generate correct HTML structure.
- **No regression**: The existing `else` branches for same-type list handling are preserved. The new code only activates for cross-type nesting scenarios.
- **CI status**: All e2e and integration tests pass across all platforms.
- **Test coverage**: 4 new test cases with expected HTML output covering the main nesting permutations.

### Concern
- The restructuring logic (removing from parent, creating new list wrapper, inserting after) is somewhat complex. If a deeply nested structure has many indent levels with alternating types, the cascading logic could theoretically produce unexpected results — but the 3-level test case covers this.

### Verdict
✅ **Safe to approve** — well-tested fix for a real formatting bug. The logic correctly handles the restructuring needed for mixed-type nested lists.

**Review by potatowagon — COMMENTED** [review]
## Review — Fix Nested Mixed-Type List Structure During Markdown Import

**Assessment: Looks good to land** ✅

### What I verified:

1. **Bug fix logic**: When importing markdown with nested lists of different types (e.g., bullet under ordered, or ordered under bullet), the previous code appended all items to the same parent list regardless of type. This fix detects type mismatches and creates a properly-typed sub-list, wrapping it in a new ListItemNode and inserting it correctly in the hierarchy.

2. **Key logic check**: The indent-based path (`indent > 0 && $isListNode(previousNode)`) now correctly appends to the previous list node. When a type mismatch is detected post-indentation, the code removes the item from the wrong-type parent, creates a new list of the correct type, wraps it in a ListItemNode, and inserts after the parent wrapper. Empty parent lists are cleaned up.

3. **Test coverage**: 4 new test cases covering: bullet→ordered nesting, ordered→bullet nesting, 3-level mixed nesting (ul→ol→ul), and mixed sublists under the same parent.

4. **CI status**: Full CI suite green (39 checks pass).

5. **Risk**: Medium — modifies the list import path which is sensitive to edge cases. However, the tests demonstrate correct behavior for the tricky nested cases, and the full e2e suite passing validates no regressions in existing list behavior.

_— via [Navi](https://navibot.dev?utm_source=github&utm_medium=pr_review&utm_campaign=8283&ref=potatowagon) on behalf of potatowagon_

**Review by etrepum — DISMISSED** [review]

**Review by etrepum — APPROVED** [review]
